SDL/LST.cc

```diff
@@ -0,0 +1,681 @@
+                            const std::vector<float> see_dxy,
+                            const std::vector<float> see_dz,
+                            const std::vector<float> see_ptErr,
+                            const std::vector<float> see_etaErr,
+                            const std::vector<float> see_stateTrajGlbX,
+                            const std::vector<float> see_stateTrajGlbY,
+                            const std::vector<float> see_stateTrajGlbZ,
+                            const std::vector<float> see_stateTrajGlbPx,
+                            const std::vector<float> see_stateTrajGlbPy,
+                            const std::vector<float> see_stateTrajGlbPz,
+                            const std::vector<int> see_q,
+                            const std::vector<unsigned int> see_algo,
+                            const std::vector<std::vector<int>> see_hitIdx,
+                            const std::vector<unsigned int> ph2_detId,
+                            const std::vector<float> ph2_x,
+                            const std::vector<float> ph2_y,
+                            const std::vector<float> ph2_z) {
+  unsigned int count = 0;
+  auto n_see = see_stateTrajGlbPx.size();
+  std::vector<float> px_vec; px_vec.reserve(n_see);
+  std::vector<float> py_vec; py_vec.reserve(n_see);
+  std::vector<float> pz_vec; pz_vec.reserve(n_see);
+  std::vector<unsigned int> hitIndices_vec0; hitIndices_vec0.reserve(n_see);
+  std::vector<unsigned int> hitIndices_vec1; hitIndices_vec1.reserve(n_see);
+  std::vector<unsigned int> hitIndices_vec2; hitIndices_vec2.reserve(n_see);
+  std::vector<unsigned int> hitIndices_vec3; hitIndices_vec3.reserve(n_see);
+  std::vector<float> ptIn_vec;   ptIn_vec.reserve(n_see);
+  std::vector<float> ptErr_vec;  ptErr_vec.reserve(n_see);
+  std::vector<float> etaErr_vec; etaErr_vec.reserve(n_see);
+  std::vector<float> eta_vec;    eta_vec.reserve(n_see);
+  std::vector<float> phi_vec;    phi_vec.reserve(n_see);
+  std::vector<int> charge_vec;    charge_vec.reserve(n_see);
+  std::vector<float> deltaPhi_vec; deltaPhi_vec.reserve(n_see);
+  std::vector<float> trkX = ph2_x;
+  std::vector<float> trkY = ph2_y;
+  std::vector<float> trkZ = ph2_z;
+  std::vector<unsigned int> hitId = ph2_detId;
+  std::vector<unsigned int> hitIdxs(ph2_detId.size());
+
+  std::vector<int> superbin_vec;
+  std::vector<int8_t> pixelType_vec;
+  std::vector<short> isQuad_vec;
+  std::iota(hitIdxs.begin(), hitIdxs.end(), 0);
+  const int hit_size = trkX.size();
+
+  int nSeeds=0;
+  for (auto &&[iSeed, _] : iter::enumerate(see_stateTrajGlbPx))
+  {
+      nSeeds++;
+      bool good_seed_type = false;
+      if (see_algo[iSeed] == 4) good_seed_type = true;
+      if (see_algo[iSeed] == 22) good_seed_type = true;
+      if (not good_seed_type) continue;
+
+      ROOT::Math::PxPyPzMVector p3LH(see_stateTrajGlbPx[iSeed], see_stateTrajGlbPy[iSeed], see_stateTrajGlbPz[iSeed],0);
```

Comment:
This function is rewritten using the latest physics vectors by ROOT. It can be used as an example to modernize the corresponding function in the standalone as well.